modules/ROOT/pages/dataweave-troubleshoot.adoc

```diff
@@ -109,3 +115,74 @@ Trace:
 When no MIME type is set on the payload, the MIME type defaults to
 `application/java`, and the content is handled as a `String` instead of a
 JSON object.
+
+=== Stack Overflow
+
+When a function recurses too deeply, an error like the following one is thrown:
+
+[source,weave, linenums]
+----
+org.mule.runtime.api.el.ExpressionExecutionException: Stack Overflow. Max stack is 256
+----
+
+You can configure the maximum stack size by using the property `com.mulesoft.dw.stacksize`.
+
+== Output Mismatch When Undefined
+
+Unlike transformations, DataWeave expressions do not require you to define an output format because DataWeave can infer the output based on the expression and the variables you use. Occasionally, the inference process results in a mismatch between the inferred type and the expected type. To resolve this issue, you must make the output explicit. Common examples of this situation occur when:
+
+* <<xml_extraction>>
+* <<multipart_handling>>
+* <<manipulating_text>>
+
+[[xml_extraction]]
+=== Extracting Data from XML
+
+When extracting a String, for example, from an XML payload with the expression `payload.order.product.model`, DataWeave  infers an XML output based on the payload format. In such cases, an error similar to the following one occurs:
```

Comment:
HERE: a String or a `String`? or a String type?

In the Term Presentation List, I have three entries. Are they correct?

string, String type

String class

string object (Internet search for all string terms)

Reply:
Remove extra space:
```suggestion
When extracting a String, for example, from an XML payload with the expression `payload.order.product.model`, DataWeave infers an XML output based on the payload format. In such cases, an error similar to the following one occurs:
```